ts/ts/main.ts

```diff
@@ -495,3 +515,56 @@ async function handleDateSearch(event: Event): Promise<void> {
     }
   }
 }
+
+if (!$saveButton) throw new Error('$saveButton not found!');
+$saveButton.addEventListener('click', saveDate);
+
+function saveDate(): void {
+  if (!currentCelebration) {
+    return;
+  }
+  if (!$savePopUp) throw new Error('$savePopUp not found!');
+  if (!$holidayName) throw new Error('$holidayName not found!');
+  if (!$holidayName.textContent) {
+    return;
+  }
+  if (!$holidayDesc) throw new Error('$holidayDesc not found!');
+  if (!$holidayDesc.textContent) {
+    return;
+  }
+  if (!$favorite) throw new Error('$favorite not found!');
+  $savePopUp.classList.add(
+    'transition-opacity',
+    'duration-500',
+    'ease-in-out',
+    'opacity-100',
+  );
+  setTimeout(() => $savePopUp.classList.remove('opacity-100'), 1000);
+  const holidayToAdd: SavedHoliday = {
+    name: $holidayName.textContent,
+    desc: $holidayDesc.textContent,
+    favorite: $favorite.dataset.favorite === 'y',
+    imageReference: currentCelebration,
+  };
+  if (!data.holidays.some((day) => day.name === holidayToAdd.name)) {
+    data.holidays.push(holidayToAdd);
+    storeData();
+  }
+}
+
+if (!$favorite) throw new Error('$favorite not found!');
+$favorite.addEventListener('click', favoriteDate);
+
+function favoriteDate(): void {
+  if (!$favorite) throw new Error('$favorite not found!');
+  if ($favorite.dataset.favorite === 'n') {
+    $favorite.dataset.favorite = 'y';
+    $favorite.classList.remove('fa-regular');
+    $favorite.classList.add('fa-solid');
+    saveDate();
+  } else {
+    $favorite.dataset.favorite = 'n';
+    $favorite.classList.remove('fa-solid');
+    $favorite.classList.add('fa-regular');
+  }
+}
```

Comment:
You would also want to modify your favorites holiday data on top of this. It's possible that the user would want to favorite the holiday after saving and so it should also update your current `data` object and then store it to `localStorage`.